clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus_test.go

```diff
@@ -565,7 +573,10 @@ func TestLogging(t *testing.T) {
 			gatherer := &apiResponsivenessGatherer{}
 			config := &measurement.Config{}
 
-			gatherer.Gather(executor, time.Now(), config)
+			_, err := gatherer.Gather(executor, time.Now(), config)
+			if err != nil {
+				t.Logf("cant gather config %v", err)
```

Comment:
err doesn't mean we couldn't gather config. It may mean violation error.

Actually looking into we can do much better, i.e. check if this is violation error and fail if not:
https://github.com/kubernetes/perf-tests/blob/345ac7aeda560a246de34d5ce89419b7eca22a05/clusterloader2/pkg/errors/metric_violation_error.go#L37

So please change to:
```
if err != nil && !errors.IsMetricViolation(err) {
  t.Errorf("error while gathering results: %v", err)
}
```

Reply:
[Also please squash all commits]

Reply:
done!